### PR TITLE
Feature/more steps

### PR DESCRIPTION
**Pull request by lapistano**

### CODE REVIEW

File: src/Command/CloudFoundry/PushCommand.php [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Process\ProcessBuilder;
+
+/**
+ * @author   List of contributors <https://github.com/libgraviton/graviton/graphs/contributors>
+ * @license  http://opensource.org/licenses/gpl-license.php GNU Public License
+ * @link     http://swisscom.ch
+ */
+final class AuthCommand extends AbstractCommand
```
Comment (hairmare) [inline src/Command/CloudFoundry/PushCommand.php]: Class name and Filename do not match PSR0. Something is fishy here...

**Comment by lapistano** [comment]
@hairmare think I am finished now.

**Comment by hairmare** [comment]
Could you point me in the direction of the tests that test the actual blue/green deploy in ``src/Command/CloudFoundry/DeployCommand.php``? I can't seem to find them and code-coverage concurs.

Next, ``bin/deploy`` only seems to find ``graviton:deployment:cf:checkApplication`` at the moment. It should also be called ``check_app`` due to most commands not using camel case and due to cloudfoundries notion of calling what we push app and not application (ie. this like ``cf apps`` and ``cf app``).

Sadly that doesn't really matter since the method is worthless. It simply replicates what I can do with a simple ``cf app <app-name>-green || cf app <app-name>-blue``. For it to be of any use it would need to do blue/green detection and tell us which of them is running (or error if none is running).

I still need to do a fully review on the code since I'm absolutely sure it still has way to many bugs. One of my main issues is that reviewing how the actual call to ``symfony/process`` is built and called is rather complex and in my opinion way to convoluted.

One thing that I am expecting from testing the actual green/blue deploy class is to see how errors are handled. The original bash script does abort at the right stage if things go haywire and it also has some informational output regarding when an app is pushed for the first time.

**Comment by hairmare** [comment]
I just fixed the map-route call in 6cddbb8f43b4f2a81cece49c813764c3764bab4e and c07f95655989c46e8b50e773b37809c16e8ff090

There are still some things that need fixing:
* [ ] on initial deploys unmapping of green routes should not be considered a failure
* [ ] on subsequent runs the code should deploy to blue/green in an alternating fashion

Please consider this command:

```bash
bin/deploy graviton:deployment:cf:deploy "test-${USER}"
```

I always get the following for each run:

```
[Symfony\Component\Process\Exception\ProcessFailedException]                           
  The command "'cf' 'unmap-route' 'test-hairmare-green' 'nova.scapp.io' '-n' 'test-hairmare'" failed.  
  Exit Code: 1(General error)                                                            
  Output:                                                                                
  ================                                                                       
  FAILED                                                                                 
  App test-hairmare-green not found                                                             
  Error Output:                                                                          
  ================       
```

**Comment by lapistano** [comment]
Fixed »unmap-route« error in d386c35c718d16db91c2e1aa0b39a3274648f1c8

**Comment by hairmare** [comment]
It looks like the implemented features are mostly in working order now. I was able to do an initial deploy and multiple redeploys with the example from my last comment.

I just noticed that this wants to merge into ``feature/symfony_config_for_lib``. Sorry about not speaking up about this earlier, I can merge it as proposed ASAP, but it'll probably still need some cleanup for merging into ``develop``.

We should probably merge this as it stands and I'll create a new PR to pull the whole shebang from  ``feature/symfony_config_for_lib`` into ``develop``. I'll then add the issues I've found so far to that PR as tasks. @lapistano let me know if that works for you...

**Comment by lapistano** [comment]
superseeded by #7

**Pull request closed without merging**